src/number/n_algExt.jl

```diff
@@ -0,0 +1,312 @@
+
+function modulus(a::N_AlgExtField)
+   return a(libSingular.transExt_GetMinpoly(a.ptr))
+end
+
+###############################################################################
+#
+#   Canonicalisation
+#
+###############################################################################
+
+canonical_unit(x::n_algExt) = x
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function show(io::IO, F::N_AlgExtField)
+   print(IOContext(io, :compact => true), "Residue field of ", base_ring(F), " modulo ", modulus(F))
+end
+
+function AbstractAlgebra.expressify(n::n_algExt; context = nothing)::Any
+   # TODO this easy method might not be the best
+   libSingular.StringSetS("")
+   libSingular.n_Write(n.ptr, parent(n).ptr, false)
+   s = libSingular.StringEndS()
+   e = Meta.parse(s)
+   if !isa(e, Expr)
+      return e
+   elseif e.head == :incomplete
+      return s
+   elseif e.head == :call && length(e.args) == 3 && e.args[1] == :/
+      e.args[1] = ://
+      return e
+   else
+      return e
+   end
+end
+
+function show(io::IO, n::n_algExt)
+   libSingular.StringSetS("")
+   libSingular.n_Write(n.ptr, parent(n).ptr, false)
+   m = libSingular.StringEndS()
+   print(io, m)
+end
+
+###############################################################################
+#
+#   Unary functions
+#
+###############################################################################
+
+function -(x::n_algExt)
+    C = parent(x)
+    ptr = libSingular.n_Neg(x.ptr, C.ptr)
+    return C(ptr)
+end
+
+###############################################################################
+#
+#   Arithmetic functions
+#
+###############################################################################
+
+function +(x::n_algExt, y::n_algExt)
+   c = parent(x)
+   p = libSingular.n_Add(x.ptr, y.ptr, c.ptr)
+   return c(p)
+end
+
+function -(x::n_algExt, y::n_algExt)
+   c = parent(x)
+   p = libSingular.n_Sub(x.ptr, y.ptr, c.ptr)
+   return c(p)
+end
+
+function *(x::n_algExt, y::n_algExt)
+   c = parent(x)
+   p = libSingular.n_Mult(x.ptr, y.ptr, c.ptr)
+   return c(p)
+end
+
+###############################################################################
+#
+#   Ad hoc arithmetic functions
+#
+###############################################################################
+
++(x::n_algExt, y::Integer) = x + parent(x)(y)
+
++(x::Integer, y::n_algExt) = parent(y)(x) + y
+
+-(x::n_algExt, y::Integer) = x - parent(x)(y)
+
+-(x::Integer, y::n_algExt) = parent(y)(x) - y
+
+*(x::n_algExt, y::Integer) = x*parent(x)(y)
+
+*(x::Integer, y::n_algExt) = parent(y)(x)*y
+
+###############################################################################
+#
+#   Comparison
+#
+###############################################################################
+
+function ==(x::n_algExt, y::n_algExt)
+    return libSingular.n_Equal(x.ptr, y.ptr, parent(x).ptr)
```

Comment:
Does this need a prent check first?

Reply:
None of the other number types check the parents. I will open an issue if this is a problem.